web/src/integTest/java/com/microsoft/applicationinsights/azuretest/CompareTelemetryWithAzureExportTests.java

```diff
@@ -0,0 +1,342 @@
+            String msgDocType = messageContentAsJson.getString("DocumentType");
+
+            if(msgDocType.equals(docType.toString())) {
+                blobUris.add(messageContentAsJson.getString("BlobUri"));
+            }
+        }
+
+        System.out.println("Got " + blobUris.size() + " blob URIs with document type " + docType.toString());
+        return blobUris;
+    }
+
+    /**
+     * Converts a string to multiple JSON objects
+     * @param jString The string to convert
+     * @return ArrayList of JSON objects
+     */
+    private ArrayList<JSONObject> getJsonsFromString(String jString) throws JSONException {
+        System.out.println("Extracting JSON objects from string");
+        String[] jsonStrings = jString.split("\n");
+        ArrayList<JSONObject> jsonObjects = new ArrayList<JSONObject>();
+
+        for(String s : jsonStrings) {
+            jsonObjects.add(new JSONObject(s));
+        }
+
+        System.out.println("Got " + jsonObjects.size() + " JSON objects");
+        return jsonObjects;
+    }
+
+    /**
+     * Converts JSON object to TelemetryItem by it's document type
+     * @param docType The document type of the telemetry
+     * @param json The JSON object
+     * @return A TelemetryItem
+     * @throws Exception
+     */
+    private TelemetryItem getTelemetryItemFromJson(DocumentType docType, JSONObject json) throws Exception {
+        if (docType == DocumentType.Requests) {
+            return getRequestTelemetryItemFromJson(json);
+        }
+
+        if (docType == DocumentType.PerformanceCounters) {
+            return getPerformanceCounterTelemetryItemFromJson(json);
+        }
+
+        throw new Exception("Can't find document type: " + docType.toString());
+    }
+
+    /**
+     * Converts JSON object to Request TelemetryItem
+     * @param json The JSON object
+     * @return A TelemetryItem
+     */
+    private TelemetryItem getRequestTelemetryItemFromJson(JSONObject json) throws URISyntaxException, JSONException {
+        System.out.println("Converting JSON object to telemetry item with document type " + DocumentType.Requests.toString());
+        String address       = json.getJSONArray("request").getJSONObject(0).getString("url");
+        Integer port         = json.getJSONArray("request").getJSONObject(0).getJSONObject("urlData").getInt("port");
+        Integer responseCode = json.getJSONArray("request").getJSONObject(0).getInt("responseCode");
+        String userAgent     = json.getJSONObject("context").getJSONObject("device").getString("userAgent");
+
+        JSONArray parameters = json.getJSONArray("request").getJSONObject(0).getJSONObject("urlData").getJSONArray("queryParameters");
+        Hashtable<String, String> queryParameters = new Hashtable<String, String>();
+        for (int i = 0; i < parameters.length(); ++i) {
+            String name  = parameters.getJSONObject(i).getString("parameter");
```

Comment:
Please store `parameters.getJSONObject(i)` in a local variable and give it a meaningful name